File: Backend/AstroGame.Shared/Models/Researches/StudiedResearch.cs
using AstroGame.Shared.Models.Players;
using System;

namespace AstroGame.Shared.Models.Researches
{
    public class StudiedResearch
    {
        public Guid Id { get; set; }
        public Guid ResearchId { get; set; }
        public Guid PlayerId { get; set; }

        public uint Level { get; set; }

        public virtual Research Research { get; set; }
        public virtual Player Player { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/researches/studied_research.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/conditions/condition.dart';
import 'package:astrogame_app/models/enums/research_type.dart';
import 'package:astrogame_app/models/researches/research.dart';
import 'package:astrogame_app/models/researches/research_cost.dart';
import 'package:flutter_guid/flutter_guid.dart';
import 'package:json_annotation/json_annotation.dart';

part 'studied_research.g.dart';

@GuidConverter()
@JsonSerializable()
class StudiedResearch extends Research {
  Guid id;
  Guid researchId;
  Guid playerId;

  int level;

  StudiedResearch();

  factory StudiedResearch.fromJson(Map<String, dynamic> json) =>
      _$StudiedResearchFromJson(json);
  Map<String, dynamic> toJson() => _$StudiedResearchToJson(this);
}
